### PR TITLE
#980  Refactor Header menu logic

### LINKED ISSUE #980: se_extras.py:20-21: Reuse the logic.header.menu_qs menu...
The puzzle `974-ceadb5de` from #974 has to be resolved: 

https://github.com/fidals/shopelectro/blob/4da0030ab4692c401c4ff0cbcbacf947275ea369/shopelectro/templatetags/se_extras.py#L20-L21

The puzzle was created by duker33 on 30-Sep-19. 

Estimate: 60 minutes,  role: DEV. 

If you have any technical questions, don't ask me, submit new tickets instead. The task will be \"done\" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

### PR DESCRIPTION
**Pull request by duker33**
Closes #980

### CODE REVIEW

File: shopelectro/logic/header.py [hunk 1]
```diff
@@ -10,14 +10,14 @@ def menu_qs() -> pages_models.PageQuerySet:
...
         .filter(
             Q(slug__in=settings.HEADER_LINKS['add'])
-            & ~Q(slug__in=settings.HEADER_LINKS['exclude'])
             | (
                 # @todo #974:30m  Optimize the header menu query.
                 #  Fetch catalog page for the header menu at the same query.
                 #  root category pages.
                 Q(parent=pages_models.CustomPage.objects.filter(slug='catalog'))
                 & Q(related_model_name=models.Category._meta.db_table)
                 & Q(parent=pages_models.CustomPage.objects.filter(slug='catalog'))
```
Comment (artemiy312) [inline shopelectro/logic/header.py]: It seems this condition duplicate [that one](https://github.com/fidals/shopelectro/pull/982/files#diff-b62fd94a6e6e9496d6a434a9aad0a060R17)

File: shopelectro/logic/header.py [hunk 2]
```diff
@@ -10,14 +10,14 @@ def menu_qs() -> pages_models.PageQuerySet:
         pages_models.Page.objects.active()
```
Comment (artemiy312) [inline shopelectro/logic/header.py]: Please, provide few words about menu content and this query in doc-string

File: shopelectro/settings/base.py [hunk 3]
```diff
@@ -465,6 +465,7 @@ def get_robots_content():
     ],
 }
 
+# add/exclude logic works only for root items
```
Comment (artemiy312) [inline shopelectro/settings/base.py]: It seems that the `add` logic can work with any elements. Please let's cover it in tests

**Comment by duker33** [comment]
tests pass locally, but not at CI https://ci.fidals.com/fidals/shopelectro/1720/9
I merge it

**Pull request merged**